test/test_assign_port.cpp

```diff
@@ -1,208 +1,91 @@
-#include "assign_port.h"
+#include "verilator_assign_port.h"
 #include "verilog_int.h"
 #include <gtest/gtest.h>
 
 using namespace std;
 using namespace verilog;
-
-struct TestAssignPort : public ::testing::Test {
-  CData cdata; // u8
-  SData sdata; // u16
-  IData idata; // u32
-  QData qdata; // u64
-  VlWide<4> data128;
-  VlWide<8> data256;
-
-  TestAssignPort() {
-    cdata = -1;
-    sdata = -1;
-    idata = -1;
-    qdata = -1;
-    for (size_t i = 0; i < 4; i++) {
-      data128[i] = -1;
-    }
-    for (size_t i = 0; i < 8; i++) {
-      data256[i] = -1;
-    }
-  }
+using namespace verilog::verilator;
+
+TEST(TestAssignPort, Read) {
+  VL_IN8(ver5,5,0);
+  VL_IN16(ver10,10,0);
+  VL_IN(ver20,20,0);
+  VL_IN64(ver40,40,0);
+  VL_INW(ver90,90,0,3);
+  VL_INW(ver100,100,0,4);
+  vuint<5> v5;
+  vuint<10> v10;
+  vuint<20> v20;
+  vuint<40> v40;
+  vuint<90> v90;
+  vuint<100> v100;
+  ver5 = 3;
+  ver10 = 1000;
+  ver20 = 1000000;
+  ver40 = 0xab12345678u;
+  ver90.m_storage[0] = 0x12345678u;
+  ver90.m_storage[1] = 0x55665566u;
+  ver90.m_storage[2] = 0xfu;
+  ver100.m_storage[0] = 0x12345678u;
+  ver100.m_storage[1] = 0x55665566u;
+  ver100.m_storage[2] = 0xffffffffu;
+  ver100.m_storage[3] = 0x1u;
+
+  read_port(v5, ver5);
+  read_port(v10, ver10);
+  read_port(v20, ver20);
+  read_port(v40, ver40);
+  read_port(v90, ver90);
+  read_port(v100, ver100);
+
+  EXPECT_EQ(v5, 3);
+  EXPECT_EQ(v10, 1000);
+  EXPECT_EQ(v20, 1000000);
+  EXPECT_EQ(v40, 0xab12345678llu);
+  EXPECT_EQ(to_hex(v90), "F5566556612345678");
+  EXPECT_EQ(to_hex(v100), "1FFFFFFFF5566556612345678");
```

Comment:
Suggest: need a test with odd number of VlWide<num_word>. L27-L29 in verilator_assign_port.h are not tested.